driver-core/src/main/com/mongodb/ReadPreference.java

```diff
@@ -48,6 +48,45 @@
     ReadPreference() {
     }
 
+    /**
```

Comment:
All these options are only supported on `TaggableReadPreference`, so let's just define these methods on that class.  

Generally, prefer compiler checks to runtime checks.

Reply:
See above.  I had to rescind this request.